ludwig/features/image_feature.py

```diff
@@ -394,39 +450,64 @@ def add_feature_data(
             lambda row: get_abs_path(src_path, row) if isinstance(row, str) and not has_remote_protocol(row) else row,
         )
 
-        (
-            should_resize,
-            width,
-            height,
-            num_channels,
-            user_specified_num_channels,
-            average_file_size,
-        ) = ImageFeatureMixin._finalize_preprocessing_parameters(preprocessing_parameters, abs_path_column)
-
-        metadata[name][PREPROCESSING]["height"] = height
-        metadata[name][PREPROCESSING]["width"] = width
-        metadata[name][PREPROCESSING]["num_channels"] = num_channels
-
-        read_image_if_bytes_obj_and_resize = partial(
-            ImageFeatureMixin._read_image_if_bytes_obj_and_resize,
-            img_width=width,
-            img_height=height,
-            should_resize=should_resize,
-            num_channels=num_channels,
-            resize_method=preprocessing_parameters["resize_method"],
-            user_specified_num_channels=user_specified_num_channels,
-        )
+        # determine if specified encoder is a torchvision model
+        torchvision_model_id = f"{feature_config[ENCODER].get(TYPE)}-{feature_config[ENCODER].get('model_variant')}"
+        torchvision_parameters = torchvision_model_registry.get(torchvision_model_id)
 
-        # TODO: alternatively use get_average_image() for unreachable images
-        default_image = get_gray_default_image(num_channels, height, width)
+        if torchvision_parameters:
+            # torchvision_parameters is not None
```

Comment:
nit: Factor these blocks out into two separate methods, one for torchvision and one 
 for Ludwig transformations, i.e. `set_torchvision_image_preprocessing_parameters()` and `set_image_preprocessing_parameters()`.